http/query.go

```diff
@@ -259,18 +223,27 @@ func (r QueryRequest) proxyRequest(now func() time.Time) (*query.ProxyRequest, e
 	// Query is preferred over spec
 	var compiler flux.Compiler
 	if r.Query != "" {
-		compiler = lang.FluxCompiler{
-			Query: r.Query,
-		}
-	} else if r.AST != nil {
-		var err error
-		r.Spec, err = toSpec(r.AST, now)
+		pkg, err := flux.Parse(r.Query)
 		if err != nil {
 			return nil, err
 		}
-		compiler = lang.SpecCompiler{
-			Spec: r.Spec,
+		c := lang.ASTCompiler{
+			AST: pkg,
+			Now: now,
+		}
+		if r.Extern != nil {
+			c.PrependFile(r.Extern)
 		}
+		compiler = c
+	} else if r.AST != nil {
+		c := lang.ASTCompiler{
+			AST: r.AST,
+			Now: now,
+		}
+		if r.Extern != nil {
+			c.PrependFile(r.Extern)
```

Comment:
do all compilers now have PrependFile?  If so, perhaps this could be done after all flavors of the compiler have been constructed.  Decreasing the number of if statements is the goal.

Reply:
@goller `PrependFile` is unique to this compiler (ASTCompiler). This seemed like the best place for this check since these extra statements are optional in terms of the HTTP API.